### PR TITLE
feat(policies): add QUANTILE normalization mode (q01/q99 robust scaling)

### PR DESCRIPTION
**Pull request by shuheng-liu**
## What this does

Adds a `QUANTILE` normalization mode — robust q01/q99 scaling — so heterogeneous community mixtures (like the curated SO101 era partitions from #477) can be normalized without either of the existing modes' failure cases. (🗃️ Feature)

**Why.** For community-scale mixtures, `MIN_MAX` is an extreme-value statistic — a single glitch frame in one dataset sets that dataset's entire scale — and `MEAN_STD` conflates task motion spread with units (a dataset whose task moves little gets a tiny std, blowing up its normalized values). LeRobot v3.0 datasets already ship `q01`/`q99` in `meta/stats.json` (verified present in all 406 datasets of the SO101 era-3 partition); this PR makes them usable end-to-end.

**What changed:**

- **`NormalizationMode.QUANTILE`** (`configs/types.py`): maps `[q01, q99] → [-1, 1]`. Arithmetic is identical to `MIN_MAX` but on the quantile buffers; values outside the quantile range extend beyond ±1 (no clamping). The zero-range snap guard (constant/padding dims) carries over unchanged.
- **Per-row fallback** (`policies/normalize.py`): datasets whose stats predate quantile support (v3.0 repos converted from v2.1 carry only min/max/mean/std) fall back to `min`/`max` for that dataset's row, with one aggregated warning. A row with neither raises a clear `KeyError`.
- **Norm-head aggregation** (`datasets/compute_stats.py`): `aggregate_feature_stats` now carries `q01`/`q99` through multi-dataset norm heads as the count-weighted mean of contributor quantiles (exact pooled quantiles are not recoverable from per-dataset quantiles; the weighted mean is the standard approximation and stays robust to a single contributor's extremes). Non-finite entries are masked per dim, mirroring the mean/variance handling.
- **Stats standardization** (`datasets/dataset_mixture.py`): quantile keys are now padded to `max_state_dim`/`max_action_dim` alongside mean/std/min/max (previously they passed through unpadded, which would shape-mismatch at buffer creation). State and actions are padded independently in case a converted dataset carries quantiles on only one feature.
- **`_inject_stats`** (`policies/pretrained.py`): stat names now come from the shared `stat_names_for_mode` helper (with the same fallback resolution) instead of a hardcoded `MEAN_STD`/`MIN_MAX` pair, so stats injection works for the new mode.

**Opt-in only.** No default `normalization_mapping` changes; existing `MEAN_STD`/`MIN_MAX` numerics are untouched (the `MIN_MAX` branch was refactored to share code with `QUANTILE` but performs the identical ops on the identical buffers). Enable via config override, e.g. `--policy.normalization_mapping '{"VISUAL": "IDENTITY", "STATE": "QUANTILE", "ACTION": "QUANTILE"}'`.

## How it was tested

- New unit tests (all CPU):
  - `tests/policies/test_normalize_per_dataset.py`: QUANTILE per-row selection (each sample scaled by its own dataset's q01/q99), Normalize→Unnormalize round-trip, min/max fallback for quantile-less rows (with warning), and a clear `KeyError` when neither quantiles nor extremes exist.
  - `tests/datasets/test_compute_stats.py`: count-weighted quantile aggregation, per-contributor fallback to same-side extremes, and no quantile keys emitted when no contributor has them.
- Full affected test dirs: `pytest tests/policies/ tests/datasets/ -m "not gpu and not network" -n auto` → **1270 passed, 9 skipped**.
- End-to-end sanity on real Hub stats: built `Normalize`/`Unnormalize` with QUANTILE from actual SO101 community `meta/stats.json` snapshots — per-dataset heads round-trip with max error ~4e-6, and a norm head aggregated across datasets carries `q01`/`q99` and normalizes real state values into the expected few-unit range.
- `pre-commit run --files` on all changed files: clean.

No `modeling_*.py`/`configuration_*.py`/optimizer/sampler/train-loop files are touched, and the new mode is opt-in (no default config uses it), so existing training behavior is unchanged.

## How to checkout & try? (for the reviewer)

```bash
pytest -sx tests/policies/test_normalize_per_dataset.py -k quantile
pytest -sx tests/datasets/test_compute_stats.py -k quantile
```

Or a 30-second interactive check:

```python
import torch
from opentau.configs.types import FeatureType, NormalizationMode, PolicyFeature
from opentau.policies.normalize import Normalize, Unnormalize

features = {"action": PolicyFeature(type=FeatureType.ACTION, shape=(2,))}
norm_map = {"ACTION": NormalizationMode.QUANTILE}
stats = [{"action": {"min": torch.tensor([-90.0, -90.0]), "max": torch.tensor([90.0, 90.0]),
                     "q01": torch.tensor([-30.0, -30.0]), "q99": torch.tensor([30.0, 30.0])}}]
norm = Normalize(features, norm_map, per_dataset_stats=stats)
print(norm({"action": torch.tensor([[30.0, -30.0]])}, torch.tensor([0])))  # -> [[1., -1.]]
```

## Checklist

- [x] I have added Google-style docstrings to important functions and ensured function parameters are typed.
- [ ] My PR includes policy-related changes.
  - [ ] If the above is checked: I have run the GPU pytests (pytest -m "gpu") and regression tests.

### CODE REVIEW

File: src/opentau/policies/pretrained.py [hunk 1]
```diff
@@ -623,9 +627,12 @@ def _inject_stats(
                 buffer = getattr(module, buffer_attr, None)
                 if buffer is None:
                     continue
-                stat_names = ("mean", "std") if norm_mode.name == "MEAN_STD" else ("min", "max")
+                stat_names = stat_names_for_mode(norm_mode)
```
[the remark this answers is not in the document]
Comment (shuheng-liu) [reply src/opentau/policies/pretrained.py:633]: Fixed in 278647a — `_stacked_num_datasets` now derives the head count from any entry of the buffer `ParameterDict` (every stat shares the `(num_datasets, ...)` leading dim by construction), so it's correct for QUANTILE and any future mode. Regression test added that runs the probe through QUANTILE-only buffers (plus a MEAN_STD control).

File: src/opentau/policies/pretrained.py [hunk 2]
```diff
@@ -623,9 +627,12 @@ def _inject_stats(
...
                 for stat in stat_names:
-                    rows = [_stat_to_float32_tensor(s[feature_key][stat]) for s in per_dataset_stats]
+                    rows = [
+                        _stat_to_float32_tensor(resolve_stat_row(s, feature_key, stat))
```
[the remark this answers is not in the document]
Comment (shuheng-liu) [reply src/opentau/policies/pretrained.py:636]: Added in 278647a — `TestInjectStatsQuantile` covers both paths: injecting q01/q99 rows into a QUANTILE policy's buffers, and injecting stats without stored quantiles (the min/max fallback via `resolve_stat_row`).

File: src/opentau/datasets/dataset_mixture.py [hunk 3]
```diff
@@ -544,12 +544,18 @@ def _to_standard_data_format(
             else:
                 raise KeyError(f"Key '{key}' not found in stats. Available keys: {list(stats.keys())}")
 
-        # pad state and action vectors
+        # pad state and action vectors (quantiles included so the QUANTILE
+        # normalization buffers see the same padded dim as mean/std/min/max;
+        # the two dicts are padded independently because converted datasets
+        # may carry quantiles on one feature but not the other)
+        padded_stat_names = ["mean", "std", "min", "max", "q01", "q10", "q50", "q90", "q99"]
```
[the remark this answers is not in the document]
Comment (shuheng-liu) [reply src/opentau/datasets/dataset_mixture.py:551]: Fixed in 278647a — the non-finite sweep now includes `q01`/`q99` (guarded as optional keys), so a NaN/Inf quantile gets the same per-repo warning as mean/std before it reaches a buffer. Also took the adjacent suggestion: `diagnose_norm_distribution.py` now supports `--norm_mode=QUANTILE` with the same min/max fallback semantics as the runtime buffers.

[the remark this answers is not in the document]
Comment (shuheng-liu) [reply src/opentau/datasets/dataset_mixture.py:551]: Added in 2a4b241 — `test_to_standard_data_format_pads_quantiles` pins all three behaviors: quantiles padded to `max_state_dim`, state/actions padded independently (actions without quantiles keeps padding its own stats), and a NaN `q01` firing the non-finite sweep warning.

File: src/opentau/scripts/diagnose_norm_distribution.py [hunk 4]
```diff
@@ -214,13 +214,21 @@ def _normalize(x: np.ndarray, stat: dict, mode: str) -> np.ndarray:
         return (x - stat["mean"]) / (stat["std"] + EPS)
     if mode == "MIN_MAX":
         return (x - stat["min"]) / (stat["max"] - stat["min"] + EPS) * 2.0 - 1.0
+    if mode == "QUANTILE":
```
[the remark this answers is not in the document]
Comment (shuheng-liu) [reply src/opentau/scripts/diagnose_norm_distribution.py:217]: Warned in 2a4b241 — `emit_corrected_stats` with `norm_mode=QUANTILE` now logs that `corrected_stats.json` carries no quantiles (not recoverable from the streaming accumulator) and that applying it downgrades heads to min/max scaling. The artifact is still written for the mean/std diagnostics it does support.

File: src/opentau/policies/normalize.py [hunk 5]
```diff
@@ -274,24 +324,32 @@ def create_stats_buffers(
...
-                        raise KeyError(
-                            f"per_dataset_stats[{i}]['{key}'] is missing stat '{name}'. "
-                            f"Available stats: {list(ds_stats[key])}."
-                        )
-                    row = _stat_to_float32_tensor(ds_stats[key][name])
+                        fallback_rows.append(i)
+                    row = _stat_to_float32_tensor(value)
                     if tuple(row.shape) != shape:
                         raise ValueError(
                             f"per_dataset_stats[{i}]['{key}']['{name}'] has shape "
```
[the remark this answers is not in the document]
Comment (shuheng-liu) [reply src/opentau/policies/normalize.py]: Fixed in 2a4b241 — the shape-mismatch error now names the resolved source stat (e.g. `'min'`) with a note that it was resolved for the requested quantile via fallback, so the message points at a key that actually exists in that row's stats.
